image/image.go

```diff
@@ -153,38 +172,105 @@ func sanitize(s string) string {
 	return strings.Replace(s, "/", "-", -1)
 }
 
-// getSchema1Manifest parses the manifest into a data structure, cleans it up, and returns it.
-// NOTE: The manifest may have been modified in the process; DO NOT reserialize and store the return value
-// if you want to preserve the original manifest; use the blob returned by Manifest() directly.
-// NOTE: It is essential for signature verification that the object is computed from the same manifest which is returned by Manifest().
-func (i *genericImage) getSchema1Manifest() (genericManifest, error) {
-	manblob, err := i.Manifest()
+// compile-time check that manifestSchema2 implements genericManifest
+var _ genericManifest = (*manifestSchema2)(nil)
+
+type manifestSchema2 struct {
+	src               types.ImageSource
+	ConfigDescriptor  descriptor   `json:"config"`
+	LayersDescriptors []descriptor `json:"layers"`
+}
+
+type descriptor struct {
+	MediaType string `json:"mediaType"`
+	Size      int64  `json:"size"`
+	Digest    string `json:"digest"`
+}
+
+func (m *manifestSchema2) String() string {
+	return ""
+}
+
+func (m *manifestSchema2) Layers() []string {
+	blobs := []string{}
+	for _, layer := range m.LayersDescriptors {
+		blobs = append(blobs, layer.Digest)
+	}
+	return blobs
+}
```

Comment:
Not having the `config` blob in here is nice, but breaks `skopeo copy`: the config won’t get copied.

I guess we need both an `Image.Layers` and `Image.Blobs` (or, perhaps rather, `LayerDigests`/`BlobDigests` to be explicit that these are not the contents, just digests).  At least the `BlobDigests` needs to exist for `Copy`; the `Layers` might not need to be exposed publicly if we do extraction (“rendering”) inside the `types.Image` abstraction.